java/src/org/openqa/selenium/grid/config/MapConfig.java

```diff
@@ -76,12 +72,19 @@ public Optional<List<String>> getAll(String section, String option) {
       Collection<?> collection = (Collection<?>) value;
       // Case when an array of map is used as config
       if (collection.stream().anyMatch(item -> item instanceof Map)) {
-        List<String> toReturn = collection.stream()
+        List<String> toReturn = new ArrayList<>();
+        collection.stream()
           .map(item -> (Map<?, ?>) item)
-          .flatMap(map -> map.entrySet().stream())
-          .map(entry -> String.format("%s=%s", entry.getKey(), entry.getValue()))
-          .sorted()
-          .collect(Collectors.toList());
+          .forEach(
```

Comment:
Maybe we can consider using the `flatMap()` construct here and just use `collect(Collectors.toList()` instead of the `for..each()` ?

Reply:
This produces a comingled list of properties from all speciified objects. I think this result would be incorrect, and it certainly doesn't match what we get from **TomlConfig**.

Using `flatMap` produces:
```
[default="brie", default="Emmental", name="Medium-hard cheese", name="soft cheese"]
```
instead of:
```
[default="brie", name="soft cheese", default="Emmental", name="Medium-hard cheese"]
```

Reply:
The assertion should probably expect a precisely specified ordered list, rather than simply confirming the presence of the expected items. The node launcher requires that the items be grouped and sorted, but this assertion doesn't ensure that the list produced **Config** object adheres to this requirement.

Reply:
I've now replaced the `containsAll` checks with `isEqualTo` and declared the "expected" list in the order that we need the **Config** objects to produce.

Reply:
For the hard requirement for list item ordering, see [NodeConfig.addDriverConfigs](https://github.com/SeleniumHQ/selenium/blob/trunk/java/src/org/openqa/selenium/grid/node/config/NodeOptions.java#L371)

Reply:
BTW - Dealing with collections of driver specifications would be easier and a lot less fragile if each driver's properties were in their own container instead of having to parse the list for instances of "display-name". The internal structure of the stored configuration maintains this level of property hierarchy, but the **Config** interface doesn't appear to allow access to individual items of object arrays.

Reply:
Perhaps add a `getArray` method to the **Config** interface to enable access to lists of properties. I suspect that this wouldn't be too difficult, and it would simplify **NodeOptions.addDriverConfigs** considerably.

Reply:
@sbabcoc - I would suggest that we track whatever enhancements you can think of as defects rather than as comments on PR so that they aren't lost and we also have a chance to vet out those enhancements with other folks.

Reply:
https://github.com/SeleniumHQ/selenium/issues/12357

Reply:
Perhaps I'll code up an implementation of this feature as my next PR.